docs/troubleshooting_guide.md

```diff
@@ -0,0 +1,358 @@
+ssh ${INSTANCE_SSH_USER:-`whoami`}@${INSTANCE_IP_ADDR_NODE_2} sudo -u root hostname
+```
+
+To enable passwordless ssh connectivity and to set up that connection manually, following actions may be performed:
+* Generate a public/private key pair:
+```bash
+[janedoe@control-host-cloudvm .ssh]$ which ssh-keygen 
+/usr/bin/ssh-keygen
+
+[janedoe@control-host-cloudvm .ssh]$ ssh-keygen -t rsa -b 4096
+Generating public/private rsa key pair.
+Enter file in which to save the key (/home/janedoe/.ssh/id_rsa): 
+Enter passphrase (empty for no passphrase): 
+Enter same passphrase again: 
+Your identification has been saved in /home/janedoe/.ssh/id_rsa.
+Your public key has been saved in /home/janedoe/.ssh/id_rsa.pub.
+The key fingerprint is:
+SHA256:<string> janedoe@control-host-cloudvm
+The key's randomart image is:
++---[RSA 4096]----+
+|.                |
+|.. o  o .        |
+| .+..* +         |
+|..........some   |
+| random stuff    |
+|ascii characters |
+|   = + +o. o .   |
+|o + O + o   .    |
+|.=.+ *o. .       |
++----[SHA256]-----+
+
+[janedoe@control-host-cloudvm .ssh]$ ls -alrt
+total 20
+-r--------. 1 janedoe janedoe  419 Nov 20 20:52 id_ed25519
+-r--------. 1 janedoe janedoe  105 Nov 20 20:52 id_ed25519.pub
+-rwxr-xr-x. 1 janedoe janedoe  578 Dec 15 18:52 known_hosts
+drwx------. 9 janedoe janedoe  237 Dec 15 19:06 ..
+-rw-r--r--. 1 janedoe janedoe  751 Dec 16 15:39 id_rsa.pub <======== Goes to BM server under ansible users home directory's: ${home}/.ssh/authorized_keys file 
+-rw-------. 1 janedoe janedoe 3243 Dec 16 15:39 id_rsa.  <======= Should be protected
+drwxrwxr-x. 2 janedoe janedoe   97 Dec 16 15:39 .
+```
+* and post the public key on to the `ansible` users HOME/.ssh/authorized_keys file using one of 2 options:
+  * if you know the password of the ansible user:
+```bash
+[janedoe@control-host-cloudvm .ssh]$ ssh-copy-id ansible@172.16.30.1
+```
+
+ * if that's unknown, then simply copy the contents of the publickey and append it into the authorized_keys file: 
+
+```bash
+[ansible@racnode1 .ssh]$ cat >> authorized_keys 
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC<more crypto> janedoe@control-host-cloudvm
+```
+
+#### 7. Failing gcscopy step
+* The following failure may be encountered during running `gcscopy.yml` playbook:
+```bash
+failed: [racnode1.orcl] (item={u'files': [u'V982068-01.zip'], u'version': u'19.3.0.0.0', u'name': u'19c_gi', u'sha256sum': [u'D668002664D9399CF61EB03C0D1E3687121FC890B1DDD50B35DCBE13C5307D2E']}) => {"ansible_loop_var": "item", "changed": false, "item": {"files": ["V982068-01.zip"], "name": "19c_gi", "sha256sum": ["D668002664D9399CF61EB03C0D1E3687121FC890B1DDD50B35DCBE13C5307D2E"], "version": "19.3.0.0.0"}, "module_stderr": "sudo: a password is required\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error", "rc": 1}
+```
+
+* Fix:
+  * Add the corresponding user `janedoe` to the OS user group wheel that has appropriate sudo privileges.
```

Comment:
How do you figure out which user needs to be added?  Should it be added on the control node or the target host?

Reply:
Added details